spyder/plugins/updatemanager/workers.py

```diff
@@ -220,77 +320,61 @@ def _check_update_available(
-                        f"{page.status_code} Client Error: {page.reason} "
-                        f"for url: {page.url}"
-                    )
-                    asset_available = False
-                    releases.remove(latest_release)
+                    asset_info = None
 
-                    latest_release = max(releases) if releases else CURRENT_VERSION
-                    update_available = CURRENT_VERSION < latest_release
+            # The asset is not available
+            logger.debug(f"Asset not available: {latest_release}")
+            releases.pop(latest_release)
 
-                    logger.debug(f"Latest release: {latest_release}")
-                    logger.debug(f"Update available: {update_available}")
+            latest_release = max(releases) if releases else CURRENT_VERSION
+            update_available = CURRENT_VERSION < latest_release
 
-        self.latest_release = latest_release
-        self.update_available = update_available
+            logger.debug(f"Latest release: {latest_release}")
+            logger.debug(f"Update available: {update_available}")
+
+        self.asset_info = asset_info
 
     def start(self):
         """Main method of the worker."""
         self.error = None
-        self.latest_release = None
-        self.update_available = False
+        self.asset_info = None
         error_msg = None
-        url = 'https://api.github.com/repos/spyder-ide/spyder/releases'
 
+        url = None
         if not is_conda_based_app():
             self.channel = "pypi"  # Default channel if not conda
             if is_conda_env(sys.prefix):
                 self.channel, channel_url = get_spyder_conda_channel()
-
-            # If Spyder is installed from defaults channel (pkgs/main), then
-            # use that channel to get updates. The defaults channel can be far
-            # behind our latest release.
-            if self.channel == "pkgs/main":
                 url = channel_url + '/channeldata.json'
-        github = "api.github.com" in url
-
-        headers = {}
-        token = os.getenv('GITHUB_TOKEN')
-        if running_in_ci() and token:
-            headers.update(Authorization=f"Bearer {token}")
 
-        logger.info(f"Checking for updates from {url}")
         try:
-            page = requests.get(url, headers=headers)
-            _rate_limits(page)
-            page.raise_for_status()
-
-            data = page.json()
-            if github:
-                # Github url
-                releases = [parse(item['tag_name']) for item in data]
-            else:
-                # Conda pkgs/main url
+            release = None
+            if url is not None:
+                logger.info(f"Getting release from {url}")
+                page = requests.get(url)
+                page.raise_for_status()
+                data = page.json()
+
+                # Conda pkgs/main or conda-forge url
+                # What about PyPi?
```

Comment:
I think you need to add and `else` to the `if is_conda_env(sys.prefix)` in line **357** above to cover this case by using the PyPI url: https://pypi.org/project/spyder

Reply:
I verified that the appropriate url should be "https://pypi.python.org/pypi/spyder/json" and included code to appropriately process the result.